packages/components/src/mermaid.tsx

```diff
@@ -1,29 +1,68 @@
-import { ReactElement, useEffect, useState } from 'react';
-import mermaid, { Config } from 'mermaid';
+import { ReactElement, useEffect, useId, useState } from 'react';
+import mermaid from 'mermaid';
 import { useTheme } from 'nextra-theme-docs';
 
-/**
- * Assign a unique ID to each mermaid svg as per requirements of `mermaid.render`.
- */
-let id = 0;
+let initialized = false;
 
 export const Mermaid = ({ chart }: { chart: string }): ReactElement => {
   const { resolvedTheme } = useTheme();
-  const [svg, setSVG] = useState('');
+  const id = useId();
+  const [svg, setSvg] = useState('');
 
+  // must use useEffect bacause mermaid needs the browser
   useEffect(() => {
-    mermaid.initialize({
-      startOnLoad: true,
-      theme: resolvedTheme as Config['theme'],
-      fontFamily: 'inherit',
-    });
-    try {
-      const renderedSvg = mermaid.render(`mermaid-svg-${(id += 1)}`, chart);
-      setSVG(renderedSvg);
-    } catch (error) {
-      console.error('Error while rendering mermaid', error);
+    // init is needed only once
+    if (!initialized) {
+      initialized = true;
+      mermaid.initialize({
+        fontFamily: 'inherit',
+      });
     }
-  }, [resolvedTheme, chart]);
+
+    // see https://mermaid-js.github.io/mermaid/#/theming?id=theme-variables-reference-table
+    const theme = JSON.stringify({
+      theme: 'base',
+      themeVariables: {
+        fontSize: '1.125rem', // text-lg
+        lineColor: resolvedTheme === 'dark' ? 'white' : 'black',
+
+        primaryColor:
+          resolvedTheme === 'dark'
+            ? '#1e293b' // slate-800
+            : '#cbd5e1', // slate-300
+        primaryTextColor: resolvedTheme === 'dark' ? 'white' : 'black',
+        primaryBorderColor:
+          resolvedTheme === 'dark'
+            ? '#cbd5e1' // slate-300
+            : '#1e293b', // slate-800
+
+        secondaryColor:
+          resolvedTheme === 'dark'
+            ? '#475569' // slate-600
+            : '#f1f5f9', // slate-100
+        secondaryBorderColor:
+          resolvedTheme === 'dark'
+            ? '#f1f5f9' // slate-100
+            : '#475569', // slate-600
+
+        tertiaryColor:
+          resolvedTheme === 'dark'
+            ? '#475569' // slate-600
+            : '#f1f5f9', // slate-100
+        tertiaryBorderColor:
+          resolvedTheme === 'dark'
+            ? '#f1f5f9' // slate-100
+            : '#475569', // slate-600
+      },
+    });
+
+    setSvg(() =>
```

Comment:
Why to use function here?

Reply:
Just wanted to avoid an unnecessary variable assignment.

Reply:
https://github.com/the-guild-org/the-guild-components/pull/517/commits/d0c4440f08e5d50aae3cbae16207f7969db89f5d